docs/source/contributor-guide/architecture.md

```diff
@@ -20,7 +20,8 @@
 # Architecture
 
 DataFusion's code structure and organization is described in the
-[Crate Documentation], to keep it as close to the source as
-possible.
+[crates.io documentation], to keep it as close to the source as
+possible. You can find the most up to date version in the [source code].
```

Comment:
What do you think about hosting the latest document generated from the source code on github pages (or other static page hoster)? Like [greptimedb.rs](https://greptimedb.rs) which is generated from https://github.com/GreptimeTeam/greptimedb/deployments/activity_log?environment=github-pages

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/5981